src/libAtomVM/bitstring.c

```diff
@@ -119,6 +135,56 @@ bool bitstring_utf8_encode(avm_int_t c, uint8_t *buf, size_t *out_size)
     return true;
 }
 
+bool bitstring_utf8_decode(int32_t *c, uint8_t *buf, size_t len, size_t *out_size)
+{
+    *c = 0;
+    if (len == 0) {
+        return false;
+    } else if ((*buf & 0xF8) == 0xF0 && len >= 4 && ((*(buf + 1) & 0xC0) == 0x80) && ((*(buf + 2) & 0xC0) == 0x80) && ((*(buf + 3) & 0xC0) == 0x80)) {
+        *c |= (*buf & 0x07) << 18;
+        *c |= (*(buf + 1) & 0x3F) << 12;
+        *c |= (*(buf + 2) & 0x3F) << 6;
+        *c |= (*(buf + 3) & 0x3F);
+        if (*c < 0 || *c > 0x10FFFF) {
```

Comment:
c cannot be less than 0 as highest bit cannot be 1.
c cannot be more than 10FFFF.

Two design choices should be made:
- Do we accept overlong encoding here, for example reject ignore values < 100000 with 4 bytes?
- How do we handle values from D800 to DFFF. Do we reject them or do we replace them with FFFD?

It seems that Erlang/OTP (25) rejects both with `unicode:characters_to_list/1`.
```erlang
% D800:
unicode:characters_to_list(<<2#11101101,2#10100000,2#10000000>>).
% overlong:
unicode:characters_to_list(<<2#11110000,2#10000000,2#10000001,2#10000101>>). 
unicode:characters_to_list(<<2#11100000,2#10000001,2#10000101>>).            
unicode:characters_to_list(<<2#11000000,2#10000101>>).                      
```

Same seems to be true with binary matching ops:
```
<<C/utf8>> = binary_to_term(term_to_binary(<<2#11110000,2#10000000,2#10000001,2#10000101>>)), C.
```

Reply:
Fixed (I think)